### PR TITLE
Add Machine ID enterprise license enforcement

### LINKED ISSUE #12915: Machine ID: Enterprise license enforcement
Similar to desktop access: for OSS, all machine ID usage is allowed. For enterprise, the license must allow it.

- Update license to include new machine ID feature
- Update Houston to allow for generating licenses with this feature.

### PR DESCRIPTION
**Pull request by timothyb89**
This adds two checks to Machine ID for license enforcement: one on initial bot create, and another on join.

Fixes #12915

### CODE REVIEW

**Comment by timothyb89** [comment]
Mild caveat that I've so far opted to not include a license check on bot renewals (only create and onboard). I figure that'll cover the 90% case without necessarily breaking people's infrastructure and am a bit wary of adding license checks in `generateUserCerts()`. It's easy enough to do if desired, though.

Tested with OSS (always enabled) and Enterprise (with + without the flag).

**Review by GavinFrazar — APPROVED** [review]
LGTM

File: lib/auth/bot_test.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	"github.com/gravitational/teleport/lib/auth/native"
+	"github.com/gravitational/teleport/lib/modules"
+	"github.com/gravitational/teleport/lib/tlsca"
+	"github.com/gravitational/teleport/lib/utils"
+	"github.com/gravitational/trace"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/crypto/ssh"
+)
+
+type botTestModules struct {
```
Comment (zmb3) [inline lib/auth/bot_test.go:35]: Can you use `modules.TestModules` here instead of implementing your own?

File: lib/auth/bot_test.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+		MachineID: false, // Explicitly turn off Machine ID
+	}
+}
+
+func TestBotCreateFeatureDisabled(t *testing.T) {
+	srv := newTestTLSServer(t)
+
+	defaultModules := modules.GetModules()
+	defer modules.SetModules(defaultModules)
+	modules.SetModules(&botTestModules{})
```
Comment (zmb3) [inline lib/auth/bot_test.go:50]: We have a `modules.SetTestModules(t, m)` that will automatically restore the modules on cleanup. I would use it here.

**Review by zmb3 — APPROVED** [review]

File: lib/auth/bot_test.go (hunk 2 above)
Comment (timothyb89) [reply lib/auth/bot_test.go]: Nice! Didn't know that existed, but it's much nicer.

**Pull request merged**